parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -214,10 +214,8 @@ public Optional<ParquetValueReader<?>> visit(
     @Override
     public Optional<ParquetValueReader<?>> visit(IntLogicalTypeAnnotation intLogicalType) {
       if (intLogicalType.getBitWidth() == 64) {
-        if (intLogicalType.isSigned()) {
-          // this will throw an UnsupportedOperationException
-          return Optional.empty();
-        }
+        Preconditions.checkArgument(
```

Comment:
We wanted to throw the exception when it is signed right? 

Now it throws when it is unsigned. check should be inverted? I will have a PR to fix it.

Reply:
oh, based on the error message. I think we should not read unsigned. So, previous check was wrong?

Reply:
I think we have lot of gaps in test coverage for parquet readers like milli-time and timestamp reading, int96 reading, all kinds of signed, unsigned reads. I will create a ticket to follow this up.